File: backend/src/main/java/com/hotspotscamp/entity/Pilot.java
package com.hotspotscamp.entity;

import java.util.UUID;

import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.Transient;
import org.springframework.data.domain.Persistable;
import org.springframework.data.relational.core.mapping.Column;
import org.springframework.data.relational.core.mapping.Table;

import com.fasterxml.jackson.annotation.JsonIgnore;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table("pilots")
public class Pilot implements Persistable<UUID> {

    @Id
    private UUID id;

    @Column("command_id")
    private UUID commandId;

    @Column("detachment_id")
    private UUID detachmentId;

    private String name;
    private Integer gunnery;
    private Integer piloting;

    @Column("as_skill")
    private Integer asSkill;

    @Column("unit_type")
    private String unitType;

    private Integer wounds;
    private Integer handicap;
    @Column("total_sp_earned")
    private Integer totalSpEarned;
    @Column("gunnery_sp_earned")
    private Integer gunnerySpEarned;
    @Column("piloting_sp_earned")
    private Integer pilotingSpEarned;
    @Column("edge_tokens_sp_earned")
    private Integer edgeTokensSpEarned;
    @Column("edge_ability_sp_earned")
    private Integer edgeAbilitySpEarned;
    @Column("edge_tokens_skill")
    private Integer edgeTokensSkill;
    @Column("edge_ability_skill")
    private Integer edgeAbilitySkill;
    @Column("edge_abilities")
    private String edgeAbilities;

    @Transient
    @Builder.Default
    @JsonIgnore
    private boolean isNew = true;

    @Override
    @JsonIgnore
    public boolean isNew() {
        return isNew || id == null;
    }

    public void setNew(boolean isNew) {
        this.isNew = isNew;
    }
}

File: frontend/src/types/pilot.ts
/*
 * HotSpots Campaigner - Battletech Mercenaries campaign management SaaS.
 * Copyright (C) 2026 Jose Ferrer
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */
import { UnitType } from './common';

/**
 * Represents a pilot or MechWarrior.
 */
export interface Pilot {
    __typename?: string;
    id: string;
    name: string;
    gunnery: number;
    piloting: number;
    asSkill: number;
    edgeTokensSkill: number | null;
    edgeAbilitySkill: number | null;
    edgeAbilities: string | null;
    unitType: UnitType;
    wounds: number;
    handicap: number;
    totalSpEarned: number;
    gunnerySpEarned: number;
    pilotingSpEarned: number;
    edgeTokensSpEarned: number;
    edgeAbilitySpEarned: number;
    detachmentId: string | null;
}

export type PilotUpdateInput = Partial<Omit<Pilot, 'id' | '__typename'>>;

// ==================== Mutation Input Variables ====================

export interface UpdatePilotVars {
    id: string;
    input: PilotUpdateInput;
}

export interface HirePilotVars {
    commandId: string;
    input: PilotUpdateInput;
}

export interface DeletePilotVars {
    pilotId: string;
}
